### ISSUE TITLE
Focus scopes are not working

### ISSUE THREAD

**Issue by HermanKirshin** [AvaloniaUI/Avalonia#10433]
**Describe the bug**

Focus scopes do not allow to keep IsFocused property state in multiple controls. Only one control has IsFocused == true at the same time.

**To Reproduce**

Create main menu and try to open it - any focused control immediately gets focus lost event and sets IsFocused = false.
See attached gifs - same app in WPF keeps IsFocused == true on control in another focus scope.

**Expected behavior**
Like it is in WPF - Control should not get focus lost events and should keep IsFocused property value inside focus scope when control in another focus scope gets focused. Thus multiple controls may have IsFocused == true at the same time.

**Screenshots**

![avalonia](https://user-images.githubusercontent.com/25755704/220659457-3e4874c8-8431-4ca8-a0b0-176f134ab11a.gif)
![wpf](https://user-images.githubusercontent.com/25755704/220659490-90535031-bd2f-43e6-b2c1-d590c7c1c393.gif)

**Desktop (please complete the following information):**

- OS: Any
- Version [11.0.0-preview5, 0.10.18]

**Additional context**

Most likely I will make commit in our 0.10.18 fork that fixes it, because such behavior is very critical for our usage, if so I can transform it into pull request later if needed, if it will not contain breaking changes.

**Comment by HermanKirshin** [comment]
I moved our fix from 0.10 to 11.0 (but had not tested after migration yet) 
https://github.com/JetBrains/Avalonia/commit/ba8a22e20ae3858cfe7b801256586387097e0792
But original pseudo-class names become confusing, because :focus-within corresponds to IsKeyboardFocusWithin property, and new property that tells if focus is inside scope is called IsFocusWithin, so I renamed :focus-within to :keyboard-focus-within, so  :keyboard-focus-within tells if control has keyboard focus and :focus-within if it is focused within scope. But it is breaking change, so probably need to find some compromise naming which will be compatible but not misleading.

**Comment by HermanKirshin** [comment]
This PR is not relevant anymore. Because IFocusScope now is not client implementable and after refactorings that allow to return focus to previous control after menu/popup close (in 13409) there is not much point in implementing smth advanced with focus scopes. 

The only thing that is still not very convenient - when in menu item handler it is needed to do smth with focused element in the main scope, not MenuItem returned by FocusManager. Luckily now it is possible with some weird workaround like this, but maybe making some legal way to do this will be useful, i.e. adding smth like this IFocusManager.GetScopeFocusedElement(IInputElement elementInScope)

```
StyledElement? GetFocusScope(IInputElement control)
{
  var c = control;
  while (c != null)
  {
    if (c is IFocusScope and Visual v && v.GetVisualRoot() is Visual { IsVisible: true })
      return v;
    c = (c as Visual)?.GetVisualParent<IInputElement>() ?? ((c as IHostedVisualTreeRoot)?.Host as IInputElement);
  }
  return null;
}
            
var mainScope = GetFocusScope(MyTextBox);
var prop = AvaloniaPropertyRegistry.Instance.FindRegistered(typeof(FocusManager), "FocusedElement");

var actualFocusedElement = mainScope.GetValue(prop);
            
```
